### ISSUE TITLE
[Question] Regarding refetching

### ISSUE THREAD

**Issue by cerino-ligutom** [vuejs/vue-apollo#973]
I'm using the composition api with `@vue/apollo-composable 4.0.0-alpha.8`

I have a pagination (connection) query and was experimenting on how to refetch the original query and start the pagination from the beginning again when a graphql variable changes (e.g. new search query or sorting changes). It seems like there are two ways currently.

1. Pass a non-reactive/ref variable on the initial query then invoke `refetch()` and pass the new variables.

2. Pass a reactive/ref variable on the initial query then upon modifying the reactive/ref variable, the refetching automatically triggers.

Is the latter intentional by design? If yes, which approach would you advise to use?

**Comment by cerino-ligutom** [comment]
After some experimenting

#1 is more explicit and pretty clear on what's going on. Quite some boilerplate to react to the variable changes and invoking `refetch`.

#2 saves a ton of code on manually invoking `refetch` and the logic for when to trigger the `refetch` which is nice but seems magical (very implicit and requires one to know that this is the case when dealing with reactive variables). Not sure if this is a normal thing and by design but I'd like to hear from the contributors.

**Comment by cerino-ligutom** [comment]
nvm, I missed it in the docs

**Comment by bbugh** [comment]
@cerino-ligutom  would you mind putting in a link to the docs so other people could find it in the future?

**Comment by cerino-ligutom** [comment]
Well, the thing is, I haven't found anything that describes the one below in the Composition API docs.

> Pass a reactive/ref variable on the initial query then upon modifying the reactive/ref variable, the refetching automatically triggers.

It's very so minimal that you won't think, or at least I didn't, **modifying the variables would trigger a refetch**.

Here's what we have as of date:

![image](https://user-images.githubusercontent.com/6721822/81184008-75cd6f80-8fe2-11ea-9f68-0ee6e90ba6f8.png)

![image](https://user-images.githubusercontent.com/6721822/81184110-94cc0180-8fe2-11ea-84e7-8613cfe90319.png)

No mention of anything about a re-fetch going to occur if a parameter changes.

Eventually I decided to check the Options API Docs and found this under the [Reactive parameters section](https://v4.apollo.vuejs.org/guide-option/queries.html#reactive-parameters).

![image](https://user-images.githubusercontent.com/6721822/81184454-02782d80-8fe3-11ea-9c8f-8209d06172c3.png)

This (red) part confirmed to me that:
1. This was by design
2. It makes sense that it'd trigger a refetch since the reactivity stuffs are now exposed with `ref`/`reactive` and as per doc above, **a refetch would occur each time a parameter changes.**

---

I went ahead and created a PR #974 in the hopes that it clarifies the side-effect to future readers.
